### PR TITLE
Tech Preview Shared Resource CSI Driver

### PR DESCRIPTION
**Pull request by adambkaplan**
Add the `CSIDriverSharedResource` feature gate to the tech preview set.
The CSI driver allows Secrets and ConfigMaps to be shared across
namespaces. This driver should only be installed on clusters that opt
into tech preview features.

### CODE REVIEW

**Comment by adambkaplan** [comment]
/cc @jitendar-singh @rolfedh @RickJWagner 

Build + Jenkins team is a no feature freeze team. We intend to make the Shared Resource CSI driver (formerly the Projected Resource CSI driver) available on a tech preview basis. Per the No FF process, we need acks from QE, Docs, and PX

**Comment by adambkaplan** [comment]
/hold

for No FF - hold can be cancelled once qe, docs, and PX sign off.

**Review by rolfedh — APPROVED** [review]

**Comment by adambkaplan** [comment]
/label docs-approved

^^^ fyi @rolfedh this is how to add the "docs approved" label ^^^

**Comment by adambkaplan** [comment]
/label px-approved

Spoke with @RickJWagner on this - he agreed that we won't need anything beyond product docs to enable end users for this CSI driver.

**Comment by adambkaplan** [comment]
/label qe-approved

QE is able to accept this.

**Review by adambkaplan — COMMENTED** [review]
/hold

The related components won't be in the OCP payload until 4.10

**Comment by gabemontero** [comment]
master branch has switched to 4.10

/hold cancel

**Review by adambkaplan — COMMENTED** [review]
@openshift/api-reviewers this PR is related to #979. We intend to use a feature gate to ensure the Shared Resource CSI driver is installed on clusters that enable tech preview features.

**Comment by adambkaplan** [comment]
Pull request which utilizes the `CSIDriverSharedResource` feature gate: https://github.com/openshift/cluster-storage-operator/pull/198

**Pull request merged**
